### ISSUE TITLE
Shellcheck finds some warnings/errors in shell scripts

### ISSUE THREAD

**Issue by dmcardle** [lowRISC/opentitan#13343]
I noticed that Shellcheck[^0] is turning up some results[^1] on OpenTitan's shell scripts. AFAICT, most of the advice seems actionable and correct.

Picking on one script as an example, Shellcheck emits an error when it sees `$@` in a string because it has surprising semantics. There is a well-documented rationale at https://www.shellcheck.net/wiki/SC2145.

```
In ./ci/scripts/check-generated.sh line 37:
        echo "Auto-generated $thing not up-to-date. Regenerate with '$@'."
                                                                     ^-- SC2145 (error): Argument mixes string and array. Use * or separate argument.
```

Do any of these scripts process untrusted input? (Are there potential security implications for any of these errors?)

Regardless, I propose that we walk through each of the errors/warnings and address root causes and add suppressions[^2] for any false-positives.

Here's a handy command for generating a colorful report:

    $ find . \( -name '*.sh' -o -name '*.bash' \) -print -exec shellcheck --severity=warning --color=always {} \; | less

[^0]: https://github.com/koalaman/shellcheck
[^1]: https://gist.github.com/dmcardle/e4a388b6b7106a2804ebf0655b1cde82
[^2]: https://www.shellcheck.net/wiki/Ignore

**Comment by dmcardle** [comment]
The report is not huge, ~500 lines (and there are fewer errors/warnings than lines). I've been planning on whittling away at these, but I haven't gotten started yet. If anyone else wants to pitch in, I'm happy to share!

**Comment by alphan** [comment]
BTW thanks for looking into this @dmcardle!

**Comment by milesdai** [comment]
Yeah I'm happy to take care of this -- it'll probably be a task for next week.

**Comment by alphan** [comment]
Thanks @milesdai! Please coordinate with @dmcardle for fixing the existing lints.

**Comment by milesdai** [comment]
Fixed in #13542
